dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/dimension/DimensionController.java

```diff
@@ -254,4 +279,37 @@ public ResponseEntity<JsonRoot> getDimensionsForDataSet(
 
     return ResponseEntity.ok(new JsonRoot("dimensions", objectNodes));
   }
+
+  /**
+   * This method performs some generic steps. It can be moved into the {@link
+   * org.hisp.dhis.webapi.controller.AbstractFullReadOnlyController} so other Controllers can use it
+   * to avoid duplication.
+   *
+   * @param rpParameters request parameters
+   * @param currentUser current {@link User} to check permissions
+   * @return {@link WebRequestData} record purely for data packaging purposes, containing {@link
+   *     WebOptions}, {@link List} of fields and {@link List} of filters
+   * @throws ForbiddenException if no permission
+   */
+  protected WebRequestData applyRequestSetup(Map<String, String> rpParameters, User currentUser)
+      throws ForbiddenException {
+
+    List<String> fields = Lists.newArrayList(contextService.getParameterValues("fields"));
+    List<String> filters = Lists.newArrayList(contextService.getParameterValues("filter"));
+
+    if (fields.isEmpty()) {
+      fields.addAll(Preset.defaultPreset().getFields());
+    }
+
+    WebOptions options = new WebOptions(rpParameters);
+
+    if (!aclService.canRead(currentUser, getEntityClass())) {
```

Comment:
I think this will always return `true` as the `DimensionalObject` interface is returned from ` getEntityClass()` and this interface doesn't have and sharing setting.

`DimensionService.getAllDimensions()` already applied ACL check for each queries so only readable objects will be returned.

Reply:
cheers, I'll remove this check. `Schema` does return `null` for `DimensionalObject` so it does always evaluate to `true`.